src/ServiceControl.Monitoring/QueueLength/QueueLengthDataStore.cs

```diff
@@ -63,16 +63,18 @@ static MonitoredEndpointValues AggregateQueueLength(List<EndpointInstanceInterva
         {
             Func<long, double> returnOneIfZero = x => x == 0 ? 1 : x;
 
-            return new MonitoredEndpointValues
+            var result = new MonitoredEndpointValues
             {
-                Average = intervals.Sum(t => t.TotalValue) / returnOneIfZero(intervals.Sum(t => t.Intervals.Length)),
                 Points = intervals.SelectMany(t => t.Intervals)
                     .GroupBy(i => i.IntervalStart)
                     .OrderBy(g => g.Key)
                     .Select(g => g.Sum(i => i.TotalValue / returnOneIfZero(i.TotalMeasurements)))
                     .ToArray()
             };
 
+            result.Average = result.Points.Length > 0 ? result.Points.Average() : 0d;
```

Comment:
I think we do. I would keep it out of this PR as a proper [test](https://github.com/Particular/ServiceControl.Monitoring/pull/41/files#diff-13946e3bafd2e391e0a09012f7372b51R1) for aggregations are included in #41.